packages/core/contracts/cross-chain-oracle/GovernorSpoke.sol

```diff
@@ -36,12 +41,15 @@ contract GovernorSpoke is Lockable, ChildMessengerConsumerInterface {
      * delegated transaction.
      */
     function processMessageFromParent(bytes memory data) public override nonReentrant() onlyMessenger() {
-        (address to, bytes memory inputData) = abi.decode(data, (address, bytes));
+        Call[] memory calls = abi.decode(data, (Call[]));
         // TODO: Consider calling this via <address>.call(): https://docs.soliditylang.org/en/v0.8.10/units-and-global-variables.html?highlight=low%20level%20call#members-of-address-types
         // to avoid inline assembly.
 
-        require(_executeCall(to, inputData), "execute call failed");
-        emit ExecutedGovernanceTransaction(to, inputData);
+        for (uint256 i = 0; i < calls.length; i++) {
+            (address to, bytes memory inputData) = (calls[i].to, calls[i].data);
+            require(_executeCall(to, inputData), "execute call failed");
```

Comment:
this forces that everything in the batch either success or fails, thereby making this atomic. nice.

Reply:
Yep!